lsp-semantic-tokens.el

```diff
@@ -449,6 +449,20 @@ If FONTIFY-IMMEDIATELY is non-nil, fontification will be performed immediately
      :cancel-token (format "semantic-tokens-%s" (lsp--buffer-uri)))))
 
 
+;;;###autoload
+(defvar-local semantic-token-modifier-cache (make-hash-table)
+  "A cache of modifier values to the selected fonts.
+This allows whole-bitmap lookup instead of checking each bit. The
+expectation is that usage of modifiers will tend to cluster, so
+we will not have the full range of possible usages, hence a
+tractable hash map.
+
+This is set as buffer-local. It should probably be shared in a
```

Comment:
yes, I guess it would make sense to store the modifiers cache as part of the workspace, cf. `lsp--semantic-tokens-initialize-workspace` (which initializes `lsp--workspace-semantic-tokens-faces` and `lsp--workspace-semantic-tokens-modifier-faces` depending on token capabilities reported by the server)

Reply:
I am leaving this as a follow-up, given that the current implementation looks for whichever first random client has a value.  Updating it could be tricky:

Line 476
```elisp
        (modifier-faces
         (when lsp-semantic-tokens-apply-modifiers
           (seq-some #'lsp--workspace-semantic-tokens-modifier-faces lsp--buffer-workspaces)))
```

Reply:
true, at some point I should probably also permit using several semhl-capable lsp servers simultaneously; though I wonder how useful that is in practice (does anyone use several servers at once, with more than one of them providing semantic highlights?)